From 440046e9220aa418ac0ffbdf126512b3341dce23 Mon Sep 17 00:00:00 2001
From: David Herrmann <dh.herrmann@gmail.com>
Date: Mon, 15 Sep 2014 17:12:41 +0200
Subject: [PATCH] terminal: fix missing hashmap_new() conversions

hashmap_new() now takes *_ops instead of individual functions. Fix up any
missing invokations of it that haven't been converted already.
---
 src/libsystemd-terminal/idev.c    |  8 ++++----
 src/libsystemd-terminal/sysview.c | 10 +++++-----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/libsystemd-terminal/idev.c b/src/libsystemd-terminal/idev.c
index d37e0313e4..8592930662 100644
--- a/src/libsystemd-terminal/idev.c
+++ b/src/libsystemd-terminal/idev.c
@@ -454,11 +454,11 @@ int idev_session_new(idev_session **out,
                         return r;
         }
 
-        s->element_map = hashmap_new(string_hash_func, string_compare_func);
+        s->element_map = hashmap_new(&string_hash_ops);
         if (!s->element_map)
                 return -ENOMEM;
 
-        s->device_map = hashmap_new(string_hash_func, string_compare_func);
+        s->device_map = hashmap_new(&string_hash_ops);
         if (!s->device_map)
                 return -ENOMEM;
 
@@ -647,11 +647,11 @@ int idev_context_new(idev_context **out, sd_event *event, sd_bus *sysbus) {
         if (sysbus)
                 c->sysbus = sd_bus_ref(sysbus);
 
-        c->session_map = hashmap_new(string_hash_func, string_compare_func);
+        c->session_map = hashmap_new(&string_hash_ops);
         if (!c->session_map)
                 return -ENOMEM;
 
-        c->data_map = hashmap_new(string_hash_func, string_compare_func);
+        c->data_map = hashmap_new(&string_hash_ops);
         if (!c->data_map)
                 return -ENOMEM;
 
diff --git a/src/libsystemd-terminal/sysview.c b/src/libsystemd-terminal/sysview.c
index bd345fa22e..2083f5a7e0 100644
--- a/src/libsystemd-terminal/sysview.c
+++ b/src/libsystemd-terminal/sysview.c
@@ -368,11 +368,11 @@ int sysview_seat_new(sysview_seat **out, sysview_context *c, const char *name) {
         if (!seat->name)
                 return -ENOMEM;
 
-        seat->session_map = hashmap_new(string_hash_func, string_compare_func);
+        seat->session_map = hashmap_new(&string_hash_ops);
         if (!seat->session_map)
                 return -ENOMEM;
 
-        seat->device_map = hashmap_new(string_hash_func, string_compare_func);
+        seat->device_map = hashmap_new(&string_hash_ops);
         if (!seat->device_map)
                 return -ENOMEM;
 
@@ -767,15 +767,15 @@ int sysview_context_new(sysview_context **out,
                         return errno > 0 ? -errno : -EFAULT;
         }
 
-        c->seat_map = hashmap_new(string_hash_func, string_compare_func);
+        c->seat_map = hashmap_new(&string_hash_ops);
         if (!c->seat_map)
                 return -ENOMEM;
 
-        c->session_map = hashmap_new(string_hash_func, string_compare_func);
+        c->session_map = hashmap_new(&string_hash_ops);
         if (!c->session_map)
                 return -ENOMEM;
 
-        c->device_map = hashmap_new(string_hash_func, string_compare_func);
+        c->device_map = hashmap_new(&string_hash_ops);
         if (!c->device_map)
                 return -ENOMEM;
 
